### DISCUSSION TITLE
SHAP dropped in performance with v0.45

### DISCUSSION THREAD

**Discussion by Alessandrovinci** [shap/shap#3965] [General]
Hello, I've been using SHAP DeepExplainer for a while with the aim of explaining the predictions of an autoencoder for anomaly detection.

I am testing the explainer on a set of anomalous instances and in all the versions up to <=0.44, the SHAP values returned made a lot of sense as they correctly identified the main factors that caused the anomaly in my points. I am sure of the accuracy of the results because I already know the reasons why this set of testing points are anomalous and I use them as ground truth of a model's results.

However, after the 0.45 big upgrade, the results no longer make sense and they are not even close as not only are the values different, but so are the features' order and magnitude as well.
 
Do you have any idea of what changed to drop the accuracy in this way? I basically can't use the package anymore and I really liked it :(

**Comment by linminhtoo** [comment]
a bit difficult to know what's happening without a minimal reproducible code sample

are you using Tensorflow or Pytorch? 
how are you setting up the DeepExplainer class and how are you putting your model through it to get the SHAP values? 

a "bruteforce" way is to look at what changed in the source code of SHAP from 0.44.1 to 0.45 (look at "files changed")
https://github.com/shap/shap/compare/v0.44.1...v0.45.0#diff-63a82e8434bcfca3c122bb66f632cfb38bffe7914940553ac70ec754b116570e 
i got this diff from https://github.com/shap/shap/releases/tag/v0.45.0 

among all the files changed, i briefly took a look at the files under shap/explainers/_deep/

however I did not see any code changes that would affect the underlying logic of these classes. it looks like the changes are either just changes to documentation or some changes in the output format (list vs numpy array). 

just to be sure, have you tried re-running your exact code, changing nothing but only reinstalling `pip install shap==0.44.1` and checking that you do indeed get the "correct" SHAP values? 

in a lot of cases (at least from my personal experience), it is actually something subtle that might have changed in your own code. 

in theory, this does not make the library unusable. you can just pin your `shap` version to 0.44.1
